DBpia.js

```diff
@@ -0,0 +1,162 @@
+{
+	"translatorID": "0c31f371-e012-4b1c-b793-f89ab1ae2610",
+	"label": "DBpia",
+	"creator": "Yunwoo Song, Philipp Zumstein",
+	"target": "^https?://[^/]+\\.dbpia\\.co\\.kr",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsi",
+	"lastUpdated": "2017-05-30 18:55:18"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	DBpia  Translator
+	Copyright © 2017 Yunwoo Song and Philipp Zumstein
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.indexOf('/Journal/ArticleDetail/')>-1) {
+		return "journalArticle";
+	} else if ((url.indexOf('/SearchResult/')>-1 || url.indexOf('/Journal/ArticleList/')>-1) && getSearchResults(doc, true)) {
+		return "multiple";
+	}
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//div[@class="titleWarp"]/a');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].href;
+		var c = i + 1;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[c] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var num = [];
+			for (var i in items) {
+				num.push(i);
+			}
+			for (var i=0; i<num.length; i++) {
+				var n = num[i];
+				var link = ZU.xpathText(doc, '//div[@class="subSearchResultList"]/div[' + n + ']/ul[1]/li[4]/dl/dt/div/a/@href')
```

Comment:
The reason I did it this way is that if the search results show books along with journal articles, I can't have the translator process the links for the books just like I can have it for journal articles. (The links for books redirect to a completely different website, and sometimes just to the main page.) So I had to use the metadata on the search result page.

However, the other metadata (like author and date) for books are not under the same node as the title and url. They are not even in sibling nodes but in like "uncle" nodes or "cousin" nodes or something. If such terminology even exits.
eg. the title is at //dl/dt/div/a
but the author is at //dl/dd[1]
then publisher at //dl/dd[2] etc.

 So I had to preserve the information from the `getSearchResult` which item was selected, and then use that information to specify the nodes with other metadata. So I had to go back all the way up to the div with "subSearchResultList" title, and then specify their count. As far as I could see on the html, that count on div was the only difference between each entry in the list.  The tags for author/date/publisher weren't marked with anything. I also think it looks very complicated. If you have better idea to simplify it, I am more than open to incorporate it.